src/sparsezoo/api/graphql.py

```diff
@@ -0,0 +1,281 @@
+    {{
+        trainingResults {args}
+            {{
+                datasetName
+                datasetType
+                displayName
+                modelId
+                primaryResult
+                recordedFormat
+                recordedUnits
+                recordedValue
+                trainingResultId
+            }}
+    }}
+"""
+
+BENCHMARK_RESULTS_GRAPHQL_BODY = """
+    {{
+        benchmarkResults {args}
+            {{
+                batchSize
+                benchmarkResultId
+                hardwareTargetDisplayName
+                hardwareTargetId
+                deviceInfo
+                displayName
+                modelId
+                numCores
+                numStreams
+                optimizationCategory
+                recordedFormat
+                recordedUnits
+                recordedValue
+                runtime
+                runtimeVersion
+                scenario                            
+            }}
+    }}
+"""
+
+
+class GraphQLAPI:
+    def __init__(self, stub: str):
+        self.stub = stub
+
+    @staticmethod
+    def search(
+        args: Dict[str, str],
+        fields: Optional[Dict[str, str]] = None,
+    ):
+        """
+        Return matching Model with respect to the args metadata
+        """
+        # if fields is None:
+        #     fields = ""
+        contemporary_stub_args = parse_stub_to_comtemporary_args(args)
+        model_input_args = parse_to_graphql_input_query(contemporary_stub_args)
+
+        url = f"{BASE_API_URL}/v2/graphql"
+        response = requests.post(
+            url=url,
+            json={
+                "query": MODEL_BODY.format(
+                    args=model_input_args,
+                    # fields=fields
+                )
+            },
+        )
+        response.raise_for_status()
+        response_json = response.json()
+
+        return response_json["data"]
+
+    @staticmethod
+    def get_file_download_url(
+        model_id: str,
+        file_name: str,
+        base_url: str = BASE_API_URL,
+    ):
+        return f"{base_url}/v2/models/{model_id}/files/{file_name}"
+
+    def get_models(self):
+        model_metadata = get_model_metadata_from_stub(self.stub)
+        return self.search(model_metadata)
+
+    def get_files(
+        self,
+        filters: Dict[str, str],
+        fields: Optional[Dict[str, str]] = None,
+        base_url: str = BASE_API_URL,
+    ):
+        contemporary_filters = parse_stub_to_comtemporary_args(filters)
+        file_input_args = parse_to_graphql_input_query(contemporary_filters)
+
+        url = f"{base_url}/v2/graphql"
+        response = requests.post(
+            url=url,
+            json={
+                "query": FILE_BODY.format(
+                    args=file_input_args,
+                    # fields=fields
+                )
+            },
+        )
+        response.raise_for_status()
+        response_json = response.json()
```

Comment:
Seems like there's some opportunity to DRY this, should you care to go in that direction